File: fe/src/app/shared/model/property.ts
import {LineChart} from "../../layout/property/property.component";

export class Property {
    _id:         string;
    city:       string;
    address:    string;
    postcode:   string;
    coordinates:Coordinates;
    extra:      Extra[];
    strExtra:   string;
    prices:     Price[];
    chart:      LineChart;

    constructor() {
        this.extra = [];
        this.prices = [];
        this.coordinates = new Coordinates();
    }
}

export class Coordinates {
    lat:    number;
    long:   number;

    constructor() { }
}

export class Extra {
    name:  string;
    value: number;

    constructor( ) { }
}

export class Price {
    data:   Date;
    value:  number;

    constructor() { }
}

File: app/models/Property.scala
package models

import reactivemongo.bson.BSONObjectID

case class Property (override val _id: Option[BSONObjectID] = newId,
                     city: String,
                     address: String,
                     postcode: String,
                     coordinates: Coordinate,
                     extra : Option[List[Extra]],
                     prices: Option[List[Price]]
                    )  extends BaseModel